include/git2/errors.h

```diff
@@ -58,6 +60,14 @@ typedef enum {
 	GIT_EMISMATCH       = -33,	/**< Hashsum mismatch in object */
 	GIT_EINDEXDIRTY     = -34,	/**< Unsaved changes in the index would be overwritten */
 	GIT_EAPPLYFAIL      = -35,	/**< Patch application failed */
+	GIT_EAGAIN          = -36,	/**< Operation would block */
+	GIT_ECANCEL         = -37,	/**< Operation was cancelled by application */
+        
+        /*
+         * Add new error codes above this comment and update GIT_EMIN
+         */
+
+	GIT_EMIN            = GIT_ECANCEL /**< Minimum error code ever generated by libgit2 */
```

Comment:
Why is this useful?

Reply:
This makes sure if the user does want to use other return codes than just GIT_EUSER, that these don't collide with libgit2 internal ones. However, as I said, I'm not particularly insistent on this.